File: COM_backend/src/main/java/com/uos/comp6239backend/service/TAuthorChapterContentService.java
package com.uos.comp6239backend.service;

import com.uos.comp6239backend.tdata.entity.*;
import com.uos.comp6239backend.utils.ResponseMap;

import java.util.List;
import java.util.Map;

/**
 * @title: TChapterContentService
 * @Author Hym
 * @Date: 2024-05-01 18:31
 * @Description:
 * @Version 1.0
 */
public interface TAuthorChapterContentService {

    ResponseMap.ResultData tryGetChapter(TAuthorChapter chapter);

    ResponseMap.ResultData updateAuthorStory(TStorys story);
    ResponseMap.ResultData insertAuthorStory(TStorys story);
    ResponseMap.ResultData insertAuthorStoryCategory(TStoryCategory storyCategory);
    ResponseMap.ResultData insertAuthorWithStory(TAuthorWithStory authorWithStory);

    ResponseMap.ResultData  deleteAuthorOptionByChapterId( Integer chapterId);
    ResponseMap.ResultData  deleteAuthorContentByChapterId( Integer chapterId);

    ResponseMap.ResultData updateAuthorChapter(TChapter chapter);
    ResponseMap.ResultData insertAuthorChapterList(List<TChapter> chapterList);
    ResponseMap.ResultData insertAuthorOptionList(List<TOption> optionList);
    ResponseMap.ResultData insertAuthorContentList(List<TContent> contentList);
    ResponseMap.ResultData tAuthorUpdateStory(TAuthorStorys values);
    ResponseMap.ResultData tAuthorStorysByAuthorId(Map<String, Object> values);
    ResponseMap.ResultData tAuthorStoryByStoryId(Map<String, Object> values);
    ResponseMap.ResultData tAuthorChapterByChapterId(Map<String, Object> values);
    ResponseMap.ResultData tAuthorChapterListByStoryId(Map<String, Object> values);
    ResponseMap.ResultData tAuthorContentListByChapterId(Map<String, Object> values);
    ResponseMap.ResultData tAuthorOptionListByChapterId(Map<String, Object> values);
    ResponseMap.ResultData tAuthorChapterByStoryIdAndChapterTitle(Map<String, Object> values);
    ResponseMap.ResultData tRootAuthorStoryByStoryId(Map<String, Object> values);
}

File: COM_project/app/src/main/java/com/codelab/basiclayouts/data/service/TAuthorChapterContentService.kt
package com.codelab.basiclayouts.data.service

import com.codelab.basiclayouts.data.ResultData
import com.codelab.basiclayouts.model.reader.readerTStorys
import com.codelab.basiclayouts.ui.uistate.author.ChapterAU
import com.codelab.basiclayouts.ui.uistate.author.StoryAU
import retrofit2.http.Body
import retrofit2.http.POST

interface TAuthorChapterContentService {

    @POST("/TAuthorChapterContentCtrl/tAuthorUpdateStory")
    suspend fun tAuthorUpdateStory(@Body values: StoryAU): ResultData<Int>


    @POST("/TAuthorChapterContentCtrl/tAuthorStorysByAuthorId")
    suspend fun tAuthorStorysByAuthorId(@Body values: Map<String, Int>): ResultData<List<StoryAU>>
    @POST("/TAuthorChapterContentCtrl/tRootAuthorStoryByStoryId")
    suspend fun tRootAuthorStoryByStoryId(@Body values: Map<String, Int>): ResultData<StoryAU>

    @POST("/TAuthorChapterContentCtrl/tAuthorStoryByStoryId")
    suspend fun tAuthorStoryByStoryId(@Body values: Map<String, Int>): ResultData<StoryAU>

    @POST("/TAuthorChapterContentCtrl/tAuthorChapterByChapterId")
    suspend fun tAuthorChapterByChapterId(@Body values: Map<String, Int>): ResultData<ChapterAU>

    @POST("/TAuthorChapterContentCtrl/tAuthorChapterListByStoryId")
    suspend fun tAuthorChapterListByStoryId(@Body values: Map<String, Int>): ResultData<ChapterAU>


}